products/libllbuild/BuildSystem-C-API.cpp

```diff
@@ -1235,9 +1235,12 @@ bool llb_buildsystem_command_interface_spawn(llb_task_interface_t ti, llb_builds
     }
 
   };
-  
-  coreti->spawn((QueueJobContext*)job_context, ArrayRef<StringRef>(arguments), ArrayRef<std::pair<StringRef, StringRef>>(environment), {true, false, StringRef((const char*)working_dir->data, working_dir->length)}, {commandCompletionFn}, new ForwardingProcessDelegate(delegate));
-  
+
+  auto forwardingDelegate = new ForwardingProcessDelegate(delegate);
```

Comment:
nit: you could also use this to manage lifetime automatically:

```c++
auto forwardingDelegate = std::make_shared<ForwardingProcessDelegate>(delegate);
```

(but you'd have to pass `forwardingDelegate.data()` to the function unless you want to refactor it to take a shared ptr)

Reply:
Good to know, thanks! I'll leave it as is for now.